API_AND_DATABASE_DESIGN.md

```diff
@@ -0,0 +1,1738 @@
+  - `duration` (Number) - Duration in minutes
+  - `status` (String) - `PENDING`, `CONFIRMED`, `COMPLETED`, `CANCELLED`, `NO_SHOW`
+  - `price` (Number) - Session price (calculated)
+  - `topic` (String) - Session topic/discussion focus
+  - `rating` (Number) - Session rating (1-5)
+  - `review` (String) - Session review text
+  - `feedbackEligible` (Boolean) - Can submit feedback (24h after completion)
+  - `feedbackSubmitted` (Boolean) - Feedback already submitted
+  - `feedbackSubmittedAt` (String) - ISO timestamp
+  - `notes` (String) - Session notes
+  - `meetingLink` (String) - Zoom/Meet link (required for virtual)
+  - `createdAt` (String) - ISO timestamp
+  - `updatedAt` (String) - ISO timestamp
+
+**GraphQL Type**:
+
+```graphql
+type Session {
+  sessionId: ID!
+  mentor: User!
+  mentee: User!
+  type: SessionType!
+  venue: Venue
+  date: String!
+  time: String!
+  duration: Int!
+  status: SessionStatus!
+  price: Float!
+  topic: String
+  rating: Float
+  review: String
+  feedbackEligible: Boolean!
+  feedbackSubmitted: Boolean!
+  feedbackSubmittedAt: String
+  notes: String
+  meetingLink: String
+  feedback: SessionFeedback
+  mentorNotes: MentorSessionNotes
+  createdAt: String!
+  updatedAt: String!
+}
+
+enum SessionType {
+  virtual
+  in_person
+}
+
+enum SessionStatus {
+  PENDING
+  CONFIRMED
+  COMPLETED
+  CANCELLED
+  NO_SHOW
+}
+```
+
+**Access Patterns**:
+- Get session by ID: Query by `PK = sessionId`
+- List sessions for mentor: Query GSI by `mentorId`, sort by `date`
+- List sessions for mentee: Query GSI by `menteeId`, sort by `date`
+- List sessions by status: Query GSI by `status`
+- List upcoming sessions: Query GSI by `date` with filter
+- List sessions by venue: Query GSI by `venueId`
+
+**Business Rules**:
+- In-person sessions require `venueId`
+- Virtual sessions require `meetingLink`
+- `feedbackEligible` becomes `true` 24 hours after completion
+- Price calculation: `basePrice * (type === 'in_person' ? 0.7 : 1.0)` (30% discount)
+- Rating bonus: In-person sessions get +0.2 rating boost
+
+---
+
+### 5. Venue (Partner Locations)
+
+**Purpose**: Partner venues for in-person sessions (libraries, co-working spaces, cafes)
+
+**DynamoDB Table**: `Venues`
+
+- **PK**: `venueId` (String) - UUID
+- **SK**: (None - single item per venue)
+
+- **Attributes**:
+  - `name` (String) - Venue name
+  - `type` (String) - `library`, `co_working`, `cafe`, `community_center`
+  - `address` (String) - Street address
+  - `city` (String) - City name
+  - `state` (String) - State/province
+  - `zipCode` (String) - Postal code
+  - `country` (String) - Country (default: "USA")
+  - `capacity` (Number) - Maximum capacity
+  - `amenities` (List of Strings) - Available amenities
+  - `image` (String) - S3 URL for venue photo
+  - `isPartner` (Boolean) - Partner venue status
+  - `partnershipTier` (String) - `bronze`, `silver`, `gold`
+  - `discountPercentage` (Number) - Discount (0-30)
```

Comment:
Maybe for cafe or co_working the discount is only on some items. For future perhaps.